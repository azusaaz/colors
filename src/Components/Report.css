.Report{

}

.report-line{
  font-size: 1.2rem;
  width: 10rem;
  display: flex;
  justify-content: space-between;
  margin: 1rem auto;
}
