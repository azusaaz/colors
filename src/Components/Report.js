import React from 'react';
import shortid from 'shortid';
import './Report.css';

const Report = ({colors}) => {
  
   // generate report from a passed object
   let report=[];
   for(let color in colors){
      report.push(
      <div className="report-line" key={shortid.generate()}>
        <span className="diamond" style={{backgroundColor: color}} key={shortid.generate()}></span>
        <span key={shortid.generate()}>{color}</span>
        <span key={shortid.generate()}>{colors[color]}</span>
      </div>
      )
   }

    return (
      <div className="Report">
         {report}
      </div>
    );
  }


export default Report;