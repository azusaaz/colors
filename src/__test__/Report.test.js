
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Report from '../Components/Report';

configure({adapter: new Adapter()});

it('expect render a report with passed props', () => {
  const testColors = {
  BLACK: 5,
  BLUE: 0,
  RED: 0,
  GREEN:0
}

expect(shallow(<Report colors={testColors}/>)).toMatchSnapshot();

});