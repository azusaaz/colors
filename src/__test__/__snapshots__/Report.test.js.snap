// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect render a report with passed props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Report
    colors={
      Object {
        "BLACK": 10,
        "BLUE": 0,
        "GREEN": 0,
        "RED": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="report-line"
        >
          <span
            className="diamond"
            style={
              Object {
                "backgroundColor": "BLACK",
              }
            }
          />
          <span>
            BLACK
          </span>
          <span>
            10
          </span>
        </div>,
        <div
          className="report-line"
        >
          <span
            className="diamond"
            style={
              Object {
                "backgroundColor": "BLUE",
              }
            }
          />
          <span>
            BLUE
          </span>
          <span>
            0
          </span>
        </div>,
        <div
          className="report-line"
        >
          <span
            className="diamond"
            style={
              Object {
                "backgroundColor": "RED",
              }
            }
          />
          <span>
            RED
          </span>
          <span>
            0
          </span>
        </div>,
        <div
          className="report-line"
        >
          <span
            className="diamond"
            style={
              Object {
                "backgroundColor": "GREEN",
              }
            }
          />
          <span>
            GREEN
          </span>
          <span>
            0
          </span>
        </div>,
      ],
      "className": "Report",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "BLACK",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "BLACK",
                }
              }
            />,
            <span>
              BLACK
            </span>,
            <span>
              10
            </span>,
          ],
          "className": "report-line",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "BLACK",
            "nodeType": "host",
            "props": Object {
              "className": "diamond",
              "style": Object {
                "backgroundColor": "BLACK",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "BLACK1",
            "nodeType": "host",
            "props": Object {
              "children": "BLACK",
            },
            "ref": null,
            "rendered": "BLACK",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "BLACK-1",
            "nodeType": "host",
            "props": Object {
              "children": 10,
            },
            "ref": null,
            "rendered": 10,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "BLUE",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "BLUE",
                }
              }
            />,
            <span>
              BLUE
            </span>,
            <span>
              0
            </span>,
          ],
          "className": "report-line",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "BLUE",
            "nodeType": "host",
            "props": Object {
              "className": "diamond",
              "style": Object {
                "backgroundColor": "BLUE",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "BLUE1",
            "nodeType": "host",
            "props": Object {
              "children": "BLUE",
            },
            "ref": null,
            "rendered": "BLUE",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "BLUE-1",
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "RED",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "RED",
                }
              }
            />,
            <span>
              RED
            </span>,
            <span>
              0
            </span>,
          ],
          "className": "report-line",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "RED",
            "nodeType": "host",
            "props": Object {
              "className": "diamond",
              "style": Object {
                "backgroundColor": "RED",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "RED1",
            "nodeType": "host",
            "props": Object {
              "children": "RED",
            },
            "ref": null,
            "rendered": "RED",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "RED-1",
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "GREEN",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "GREEN",
                }
              }
            />,
            <span>
              GREEN
            </span>,
            <span>
              0
            </span>,
          ],
          "className": "report-line",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "GREEN",
            "nodeType": "host",
            "props": Object {
              "className": "diamond",
              "style": Object {
                "backgroundColor": "GREEN",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "GREEN1",
            "nodeType": "host",
            "props": Object {
              "children": "GREEN",
            },
            "ref": null,
            "rendered": "GREEN",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "GREEN-1",
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="report-line"
          >
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "BLACK",
                }
              }
            />
            <span>
              BLACK
            </span>
            <span>
              10
            </span>
          </div>,
          <div
            className="report-line"
          >
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "BLUE",
                }
              }
            />
            <span>
              BLUE
            </span>
            <span>
              0
            </span>
          </div>,
          <div
            className="report-line"
          >
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "RED",
                }
              }
            />
            <span>
              RED
            </span>
            <span>
              0
            </span>
          </div>,
          <div
            className="report-line"
          >
            <span
              className="diamond"
              style={
                Object {
                  "backgroundColor": "GREEN",
                }
              }
            />
            <span>
              GREEN
            </span>
            <span>
              0
            </span>
          </div>,
        ],
        "className": "Report",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "BLACK",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="diamond"
                style={
                  Object {
                    "backgroundColor": "BLACK",
                  }
                }
              />,
              <span>
                BLACK
              </span>,
              <span>
                10
              </span>,
            ],
            "className": "report-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "BLACK",
              "nodeType": "host",
              "props": Object {
                "className": "diamond",
                "style": Object {
                  "backgroundColor": "BLACK",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "BLACK1",
              "nodeType": "host",
              "props": Object {
                "children": "BLACK",
              },
              "ref": null,
              "rendered": "BLACK",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "BLACK-1",
              "nodeType": "host",
              "props": Object {
                "children": 10,
              },
              "ref": null,
              "rendered": 10,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "BLUE",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="diamond"
                style={
                  Object {
                    "backgroundColor": "BLUE",
                  }
                }
              />,
              <span>
                BLUE
              </span>,
              <span>
                0
              </span>,
            ],
            "className": "report-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "BLUE",
              "nodeType": "host",
              "props": Object {
                "className": "diamond",
                "style": Object {
                  "backgroundColor": "BLUE",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "BLUE1",
              "nodeType": "host",
              "props": Object {
                "children": "BLUE",
              },
              "ref": null,
              "rendered": "BLUE",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "BLUE-1",
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "RED",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="diamond"
                style={
                  Object {
                    "backgroundColor": "RED",
                  }
                }
              />,
              <span>
                RED
              </span>,
              <span>
                0
              </span>,
            ],
            "className": "report-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "RED",
              "nodeType": "host",
              "props": Object {
                "className": "diamond",
                "style": Object {
                  "backgroundColor": "RED",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "RED1",
              "nodeType": "host",
              "props": Object {
                "children": "RED",
              },
              "ref": null,
              "rendered": "RED",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "RED-1",
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "GREEN",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="diamond"
                style={
                  Object {
                    "backgroundColor": "GREEN",
                  }
                }
              />,
              <span>
                GREEN
              </span>,
              <span>
                0
              </span>,
            ],
            "className": "report-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "GREEN",
              "nodeType": "host",
              "props": Object {
                "className": "diamond",
                "style": Object {
                  "backgroundColor": "GREEN",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "GREEN1",
              "nodeType": "host",
              "props": Object {
                "children": "GREEN",
              },
              "ref": null,
              "rendered": "GREEN",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "GREEN-1",
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
